zerver/tests/test_signup.py

```diff
@@ -1510,6 +1509,31 @@ def test_accessing_invites_in_another_realm(self) -> None:
         error_result = self.client_delete('/json/invites/' + str(prereg_user.id))
         self.assert_json_error(error_result, "No such invitation")
 
+    def test_check_user_able_to_register_completely(self) -> None:
+        email_ = self.nonreg_email("alice")
+        password = "password"
+
+        inviter = UserProfile.objects.first()
+        prereg_user = PreregistrationUser.objects.create(
+            email=email_, referred_by=inviter, realm=get_realm('zulip'))
+
+        confirmation_link = create_confirmation_link(prereg_user, 'host', Confirmation.USER_REGISTRATION)
+        registration_key = confirmation_link.split('/')[-1]
+
+        confirmation = Confirmation.objects.get(confirmation_key=registration_key)
+        prereg_user = confirmation.content_object
+        active_value = getattr(confirmation_settings, 'STATUS_ACTIVE', 1)
+
+        url = "/accounts/register/"
+        self.client_post(url, {"key": registration_key, "from_confirmation": 1, "full_name": "alice"})
+        self.assertNotEqual(prereg_user.status, active_value)
+
+        result = self.submit_reg_form_for_user(email_, password, key=registration_key)
+        self.assertEqual(result.status_code, 302)
+        prereg_user = PreregistrationUser.objects.get(
+            email=email_, referred_by=inviter, realm=get_realm('zulip'))
+        self.assertEqual(prereg_user.status, active_value)
```

Comment:
one last assertion I'd add here would be that "alice" actually has a ``UserProfile``now - so successfully registered - to ensure that this ``result`` isn't something else than we think it is.

Reply:
How do I check if Alice has a user profile?

Reply:
``get_user`` function to try to get the UserProfile with her email form the database.

Reply:
@mateuszmandera I made the changes. Please have a look. Also stashed the commits.